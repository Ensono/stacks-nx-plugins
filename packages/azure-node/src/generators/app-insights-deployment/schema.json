{
  "$schema": "http://json-schema.org/schema",
  "$id": "AppInsightsDeployment",
  "title": "",
  "type": "object",
  "properties": {
    "project": {
      "type": "string",
      "description": "The name of the project.",
      "alias": "p",
      "$default": {
        "$source": "projectName"
      },
      "x-prompt": "What is the name of the project for this component?"
    },
    "applicationinsightsConnectionString": {
      "type": "string",
      "description": "The env variable that stores the app insights connection string.",
      "alias": "e",
      "default": "APPLICATIONINSIGHTS_CONNECTION_STRING",
      "x-prompt": "What is the env variable for the application insights connection string?",
      "pattern": "^[A-Z]+(?:_[A-Z]+)*$",
      "minLength": 5,
      "maxLength": 100
    }
  },
  "required": [
    "project",
    "applicationinsightsConnectionString"
  ]
}
