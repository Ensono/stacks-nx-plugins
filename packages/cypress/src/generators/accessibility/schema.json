{
    "$schema": "http://json-schema.org/schema",
    "cli": "nx",
    "$id": "Accessibility",
    "title": "Ensono Stacks cypress Accessibility Generator",
    "description": "The @ensono-stacks/cypress:accessibility generator scaffolds your existing initialised cypress project with example Accessibility tests and config.",
    "type": "object",
    "properties": {
        "project": {
            "type": "string",
            "description": "The name of the project.",
            "alias": "p",
            "$default": {
                "$source": "projectName"
            },
            "x-prompt": "What is the name of the project for this component?"
        }
    },
    "required": ["project"]
}
