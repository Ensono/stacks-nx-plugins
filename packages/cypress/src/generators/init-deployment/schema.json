{
  "$schema": "http://json-schema.org/schema",
  "$id": "InitDeployment",
  "title": "Ensono Stacks Cypress Init Deployment Generator",
  "description": "The @ensono-stacks/cypress:init-deployment generator adds the relevant pipeline and infrastructure code to your workspace.",
  "type": "object",
  "properties": {},
  "required": []
}
