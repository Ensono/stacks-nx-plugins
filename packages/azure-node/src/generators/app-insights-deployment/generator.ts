import {
    executedDependantGenerator,
    hasGeneratorExecutedForProject,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import { readProjectConfiguration, Tree } from '@nrwl/devkit';

import { AppInsightsDeploymentGeneratorSchema } from './schema';
import { updateDockerfile } from './utils/update-dockerfile';
import { updatePipelineStagesYaml } from './utils/update-pipeline';
import { updateProjectJsonHelmUpgradeTarget } from './utils/update-targets';

// eslint-disable-next-line consistent-return
export default async function appInsightsDeploymentGenerator(
    tree: Tree,
    options: AppInsightsDeploymentGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree, options.project);

    if (
        !executedDependantGenerator(
            tree,
            'WorkspaceDeployment',
            options.project,
        )
    )
        return false;
    if (
        hasGeneratorExecutedForProject(
            tree,
            options.project,
            'AzureNodeAppInsightsDeployment',
        )
    )
        return false;

    const project = readProjectConfiguration(tree, options.project);

    // Update project.json
    updateProjectJsonHelmUpgradeTarget(project, tree, options);
    updatePipelineStagesYaml(tree, options);
    updateDockerfile(project, tree);
}
