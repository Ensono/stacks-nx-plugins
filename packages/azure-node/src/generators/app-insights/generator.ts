import {
    tsMorphTree,
    addIgnoreEntry,
    formatFiles,
    thirdPartyDependencyWarning,
    deploymentGeneratorMessage,
    hasGeneratorExecutedForProject,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    getProjects,
    joinPathFragments,
    Tree,
} from '@nx/devkit';
import { runTasksInSerial } from '@nx/workspace/src/utilities/run-tasks-in-serial';
import chalk from 'chalk';
import path from 'path';

import { AppInsightsGeneratorSchema } from './schema';
import {
    initAppInsights,
    configureAppInsights,
    startAppInsights,
} from './templates/appInsights';
import { appInsightsVersion } from '../../../utils/versions';

function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {
            applicationinsights: appInsightsVersion,
        },
        {},
    );
}

export default async function appInsightsGenerator(
    tree: Tree,
    options: AppInsightsGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree, options.project);

    if (
        hasGeneratorExecutedForProject(
            tree,
            options.project,
            'AzureNodeAppInsights',
        )
    )
        return false;

    const { applicationinsightsConnectionString, server } = options;
    const project = getProjects(tree).get(options.project);

    if (!project) {
        throw new Error(
            `No application was found with the name '${options.project}'`,
        );
    }

    const customServerPath = path.join(project.sourceRoot, server);

    // Check if custom server exist
    if (!tree.exists(customServerPath)) {
        throw new Error('No custom server found.');
    }

    const morphTree = tsMorphTree(tree);

    // Read the Node from the source file
    const customServer = morphTree.addSourceFileAtPath(customServerPath);

    const isAppInsightsImport = customServer
        .getImportDeclarations()
        .some(
            importDeclaration =>
                importDeclaration.getModuleSpecifier().getLiteralValue() ===
                'applicationinsights',
        );

    // Add appInsights to import if it's not there
    if (isAppInsightsImport) {
        throw new Error('AppInsights SDK is already in use.');
    }

    thirdPartyDependencyWarning(['@nx/next:custom-server']);

    customServer.addImportDeclaration({
        namespaceImport: 'appInsights',
        moduleSpecifier: 'applicationinsights',
    });

    const mainFunction = customServer.getFunction('main');

    // Add appInisghts statements
    mainFunction.insertStatements(
        0,
        initAppInsights(applicationinsightsConnectionString),
    );
    mainFunction.insertStatements(1, configureAppInsights(project.name));
    mainFunction.insertStatements(2, startAppInsights());

    // Add empty lines after appInisghts statements
    mainFunction.getStatements().forEach(statement => {
        if (statement.getText().includes('appInsights')) {
            statement.appendWhitespace(writer => writer.newLine());
        }
    });

    // Save changes
    customServer.saveSync();

    const serverPath = joinPathFragments(project.root, server);
    // add nrwl/next custom server to prettier ignore
    addIgnoreEntry(tree, '.prettierignore', 'next server', [serverPath]);
    // Format files excluding the server file
    await formatFiles(tree, [serverPath]);

    console.warn(
        chalk.yellow`${serverPath} has been added to .prettierignore; Amend this file to resolve linting issues.`,
    );

    // Add dependencies and install
    return runTasksInSerial(updateDependencies(tree), () =>
        deploymentGeneratorMessage(
            tree,
            `nx g @ensono-stacks/azure-node:app-insights-deployment --project ${options.project} --applicationinsightsConnectionString ${applicationinsightsConnectionString}`,
        ),
    );
}
