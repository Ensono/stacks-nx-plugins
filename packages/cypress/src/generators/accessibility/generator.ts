import {
    hasGeneratorExecutedForProject,
    isGeneratorInExecutedListForProject,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import {
    addDependenciesToPackageJson,
    formatFiles,
    Tree,
    updateJson,
    joinPathFragments,
} from '@nrwl/devkit';

import { normalizeOptions, addFiles } from '../../utils/test-utils';
import { AXECORE_VERSION, CYPRESSAXE_VERSION } from '../../versions';
import { AccessibilityGeneratorSchema } from './schema';
import { addTerminalLogging, updateCypressConfig } from './utils/update-files';

async function updateDependencies(tree: Tree) {
    return addDependenciesToPackageJson(
        tree,
        {},
        {
            'axe-core': AXECORE_VERSION,
            'cypress-axe': CYPRESSAXE_VERSION,
        },
    );
}

function updateTsConfig(tree: Tree, project: string) {
    updateJson(
        tree,
        joinPathFragments(project, 'tsconfig.json'),
        tsConfigJson => {
            const updatedProjectJson = { ...tsConfigJson };
            if (
                !updatedProjectJson.compilerOptions.types.includes(
                    'cypress-axe',
                )
            ) {
                updatedProjectJson.compilerOptions.types.push('cypress-axe');
            }
            return updatedProjectJson;
        },
    );
}

export default async function accessibilityGenerator(
    tree: Tree,
    options: AccessibilityGeneratorSchema,
) {
    verifyPluginCanBeInstalled(tree, options.project);

    const normalizedOptions = normalizeOptions(tree, options);
    if (
        hasGeneratorExecutedForProject(
            tree,
            options.project,
            'CypressAccessibility',
        )
    )
        return false;
    isGeneratorInExecutedListForProject(
        tree,
        options.project,
        'CypressInit',
        true,
    );

    // generate acessiblity files
    addFiles(
        tree,
        'files',
        __dirname,
        normalizedOptions.projectRoot,
        normalizedOptions,
    );

    // update ts config
    updateTsConfig(tree, normalizedOptions.cypressProject);

    updateCypressConfig(tree, normalizedOptions.projectRoot);

    // add terminal logging funcitonality to e2e.ts
    addTerminalLogging(tree, normalizedOptions.cypressProject);
    await formatFiles(tree);

    return updateDependencies(tree);
}
