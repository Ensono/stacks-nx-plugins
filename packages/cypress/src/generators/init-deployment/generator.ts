import {
    executedDependantGenerator,
    hasGeneratorExecutedForWorkspace,
    verifyPluginCanBeInstalled,
} from '@ensono-stacks/core';
import { formatFiles, Tree } from '@nx/devkit';

import { updateAzureDevopsStages } from './utils/update-azdevops-build';
import { updateTaskctlYaml, updateTasksYaml } from './utils/update-tasks-yamls';

// eslint-disable-next-line consistent-return
export default async function initDeploymentGenerator(tree: Tree) {
    verifyPluginCanBeInstalled(tree);

    if (!executedDependantGenerator(tree, 'WorkspaceDeployment')) return false;
    if (hasGeneratorExecutedForWorkspace(tree, 'CypressInitDeployment'))
        return false;

    updateTaskctlYaml(tree);
    updateTasksYaml(tree);
    updateAzureDevopsStages(tree);

    await formatFiles(tree);
}
