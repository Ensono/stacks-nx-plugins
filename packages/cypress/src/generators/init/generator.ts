import {
    addIgnoreEntry,
    deploymentGeneratorMessage,
    hasGeneratorExecutedForProject,
    tsMorphTree,
} from '@ensono-stacks/core';
import {
    CypressE2EConfigSchema,
    cypressE2EConfigurationGenerator,
} from '@nrwl/cypress/src/generators/cypress-e2e-configuration/cypress-e2e-configuration';
import {
    formatFiles,
    generateFiles,
    getProjects,
    offsetFromRoot,
    Tree,
    readProjectConfiguration,
    addDependenciesToPackageJson,
    updateJson,
} from '@nrwl/devkit';
import { Linter } from '@nrwl/linter';
import { runTasksInSerial } from '@nrwl/workspace/src/utilities/run-tasks-in-serial';
import path from 'path';

import { updateLintFile } from '../utils/update-files';
import { updateProjectJsonWithHtmlReport } from '../utils/update-targets';
import { CypressGeneratorSchema } from './schema';

interface NormalizedSchema extends CypressGeneratorSchema {
    projectName: string;
    e2eNameForProject: string;
    projectRoot: string;
}

function normalizeOptions(
    tree: Tree,
    options: CypressGeneratorSchema,
): NormalizedSchema {
    const project = getProjects(tree).get(options.project);

    return {
        ...options,
        projectName: project?.name as string,
        e2eNameForProject: `${project?.name as string}-e2e`,
        projectRoot: project?.root as string,
    };
}

function addFiles(
    tree: Tree,
    source: string,
    destination: string,
    options: NormalizedSchema,
) {
    const templateOptions = {
        ...options,
        offsetFromRoot: offsetFromRoot(options.projectRoot),
        template: '',
    };

    generateFiles(
        tree,
        path.join(__dirname, source),
        destination,
        templateOptions,
    );
}

function updateDependencies(tree) {
    return addDependenciesToPackageJson(
        tree,
        {},
        {
            // Needs changing
            '@nrwl/cypress': '15.9.2',
        },
    );
}

export default async function initGenerator(
    tree: Tree,
    options: CypressGeneratorSchema,
) {
    if (hasGeneratorExecutedForProject(tree, options.project, 'CypressInit'))
        return false;
    const normalizedOptions = normalizeOptions(tree, options);
    await cypressE2EConfigurationGenerator(tree, {
        project: normalizedOptions.projectName,
    });
    // update eslint.rc
    updateLintFile(tree);

    // add / remove files
    tree.delete(
        path.join(
            normalizedOptions.e2eNameForProject,
            'src',
            'support',
            'app.po.ts',
        ),
    );
    tree.delete(
        path.join(
            normalizedOptions.e2eNameForProject,
            'src',
            'e2e',
            'app.cy.ts',
        ),
    );
    addFiles(
        tree,
        path.join('files', 'e2e-folder'),
        normalizedOptions.e2eNameForProject,
        normalizedOptions,
    );
    addFiles(
        tree,
        path.join('files', 'root'),
        normalizedOptions.projectRoot,
        normalizedOptions,
    );

    // update targets
    updateProjectJsonWithHtmlReport(
        readProjectConfiguration(tree, normalizedOptions.e2eNameForProject),
        tree,
    );
    // update git ignore
    addIgnoreEntry(tree, '.gitignore', 'Cypress', ['**/test-results']);

    await formatFiles(tree);

    return runTasksInSerial(updateDependencies(tree), () =>
        deploymentGeneratorMessage(
            tree,
            'nx g @ensono-stacks/cypress:init-deployment',
        ),
    );
}
