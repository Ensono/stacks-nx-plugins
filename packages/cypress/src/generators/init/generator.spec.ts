import { tsMorphTree } from '@ensono-stacks/core';
import { addStacksAttributes } from '@ensono-stacks/test';
import { joinPathFragments, readJson, Tree } from '@nrwl/devkit';
import { createTreeWithEmptyWorkspace } from '@nrwl/devkit/testing';
import { applicationGenerator } from '@nrwl/next';
import { Schema as NextSchema } from '@nrwl/next/src/generators/application/schema';
import exp from 'constants';
import * as fs from 'fs';
import path from 'path';
import { SyntaxKind } from 'ts-morph';

import generator from './generator';
import { CypressGeneratorSchema } from './schema';

const projectName = 'test';
const projectNameE2E = `${projectName}-e2e`;

let appTree: Tree;
let options: CypressGeneratorSchema;

jest.mock('@nrwl/devkit', () => {
    const actual = jest.requireActual('@nrwl/devkit');

    return {
        ...actual,
        getProjects: jest.fn(
            () =>
                new Map([
                    [
                        'test',
                        {
                            root: '',
                            sourceRoot: `${projectName}`,
                            name: 'test',
                        },
                    ],
                ]),
        ),
    };
});

function compareToFile(fileInTree, fileToMatchAgainstPath: string) {
    const expectedFileContents = fs
        .readFileSync(path.resolve(__dirname, fileToMatchAgainstPath), 'utf-8')
        .replace(/(\r)/gm, '')
        .trim();
    const fileContents = fileInTree.getFullText().trim();
    expect(fileContents).toBe(expectedFileContents);
}

async function createNextApp(schema?: Partial<NextSchema>) {
    appTree = createTreeWithEmptyWorkspace();
    await applicationGenerator(appTree, {
        name: projectName,
        style: 'css',
        ...schema,
    });

    addStacksAttributes(appTree, options.project);
}

describe('cypress generator', () => {
    beforeEach(async () => {
        options = {
            project: projectName,
        };
        await createNextApp();
    });

    it('should resolve false if the project already exists', async () => {
        await generator(appTree, options);
        await expect(generator(appTree, options)).resolves.toBe(false);
    }, 100_000);

    describe('executedGenerators', () => {
        beforeEach(async () => {
            await generator(appTree, options);
        });

        it('should update nx.json and tag executed generator true', async () => {
            const nxJson = readJson(appTree, 'nx.json');

            expect(
                nxJson.stacks.executedGenerators.project[
                    options.project
                ].includes('CypressInit'),
            ).toBe(true);
        });

        it('should return false from method and exit generator if already executed', async () => {
            const gen = await generator(appTree, {
                ...options,
            });

            expect(gen).toBe(false);
        });
    });
});

describe('should run successfully with default options', () => {
    let project;

    beforeAll(async () => {
        options = {
            project: projectName,
        };
        await createNextApp();
        await generator(appTree, options);
        project = tsMorphTree(appTree);
    });

    it('should update the eslintrc.json', () => {
        const eslintrc = readJson(appTree, '/.eslintrc.json');
        expect(eslintrc.plugins).toContain('cypress');
        expect(eslintrc.overrides[1].extends).toContain(
            'plugin:cypress/recommended',
        );
    });

    it('should update the gitignore', () => {
        // expect .gitignore entries to be added
        const gitIgnoreFile = appTree.read('/.gitignore', 'utf-8');
        expect(gitIgnoreFile).toContain('**/test-results');
    });

    it('should update the project cypress config', () => {
        // expect cypress config to be updated
        expect(
            appTree.exists(path.join(projectNameE2E, 'cypress.config.ts')),
        ).toBeTruthy();

        const projectConfigFile = project.addSourceFileAtPath(
            `${projectNameE2E}/cypress.config.ts`,
        );
        compareToFile(
            projectConfigFile,
            './files/e2e-folder/cypress.config.ts__template__',
        );
    });

    it('should update the project.json with the html-report target', () => {
        const projectJson = readJson(
            appTree,
            joinPathFragments(projectNameE2E, 'project.json'),
        );
        expect(projectJson.targets.e2e).toBeTruthy();
        const expectedJson = {
            executor: 'nx:run-commands',
            options: {
                commands: [
                    'mochawesome-merge reports-json-file/*.json>merged-html-report.json',
                    'marge merged-html-report.json --reportDir ./ --inline',
                ],
                parallel: false,
                cwd: 'apps/next-app-e2e/test-results/downloads',
            },
            configurations: {
                ci: {
                    cwd: 'apps/../test-results/next-app-e2e/downloads',
                },
            },
        };
        expect(projectJson.targets['html-report']).toBeTruthy();
        expect(projectJson.targets['html-report']).toEqual(expectedJson);
    });

    it('should set up the example files', () => {
        expect(
            appTree.exists(
                path.join(projectNameE2E, 'src', 'e2e', 'app.cy.ts'),
            ),
        ).toBeFalsy();
        expect(
            appTree.exists(
                path.join(projectNameE2E, 'src', 'support', 'app.po.ts'),
            ),
        ).toBeFalsy();
        const filePath = path.join(
            projectNameE2E,
            'src',
            'e2e',
            'example.cy.ts',
        );
        expect(appTree.exists(filePath)).toBeTruthy();
        compareToFile(
            project.addSourceFileAtPath(filePath),
            './files/e2e-folder/src/e2e/example.cy.ts__template__',
        );
    });

    it('should update the support e2e file', () => {
        const filePath = path.join(projectNameE2E, 'src', 'support', 'e2e.ts');
        expect(appTree.exists(filePath)).toBeTruthy();
        compareToFile(
            project.addSourceFileAtPath(filePath),
            './files/e2e-folder/src/support/e2e.ts__template__',
        );
    });

    it('should update the tsconfig.json', () => {
        const configJson = readJson(
            appTree,
            joinPathFragments(projectNameE2E, 'tsconfig.json'),
        );
        expect(
            configJson.compilerOptions['allowSyntheticDefaultImports'],
        ).toEqual(true);
    });

    it('should create the base cypress configuration', () => {
        const filePath = 'cypress.config.base.ts';
        expect(appTree.exists(filePath)).toBeTruthy();
        compareToFile(
            project.addSourceFileAtPath(filePath),
            './files/root/cypress.config.base.ts__template__',
        );
    });
});
