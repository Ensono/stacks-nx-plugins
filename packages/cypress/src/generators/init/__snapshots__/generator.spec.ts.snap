// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should run successfully with default options should move the tsconfig.cy.json into the cypress directory and configure it has configured the new tsconfig.json within the cypress directory: apps/application-e2e/src/tsconfig.json 1`] = `
"{
  "extends": "../tsconfig.json",
  "compilerOptions": {
    "allowJs": true,
    "outDir": "../../dist/out-tsc",
    "module": "commonjs",
    "types": ["cypress", "node"],
    "sourceMap": false
  },
  "include": [
    "**/*.ts",
    "**/*.js",
    "../cypress.config.ts",
    "../**/*.cy.ts",
    "../**/*.cy.js",
    "../**/*.d.ts"
  ],
  "exclude": ["cypress/**/**", "cypress.config.ts"]
}
"
`;

exports[`should run successfully with default options should set up the example files: apps/application-e2e/src/fixtures/example.json 1`] = `
"{
  "name": "Using fixtures to represent data",
  "email": "hello@cypress.io",
  "body": "Fixtures are a great way to mock data for responses to routes"
}
"
`;

exports[`should run successfully with default options should set up the example files: apps/application-e2e/src/support/commands.ts 1`] = `
"declare namespace Cypress {
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  interface Chainable<Subject> {
    login(email: string, password: string): void;
  }
}

// -- This is a parent command --
Cypress.Commands.add('login', (email, password) => {
  console.log('Custom command example: Login', email, password);
});
//
// -- This is a child command --
// Cypress.Commands.add("drag", { prevSubject: 'element'}, (subject, options) => { ... })
//
//
// -- This is a dual command --
// Cypress.Commands.add("dismiss", { prevSubject: 'optional'}, (subject, options) => { ... })
//
//
// -- This will overwrite an existing command --
// Cypress.Commands.overwrite("visit", (originalFn, url, options) => { ... })
"
`;

exports[`should run successfully with default options should update the project cypress config: apps/application-e2e/cypress.config.ts 1`] = `
"import { nxE2EPreset } from '@nx/cypress/plugins/cypress-preset';

import { defineConfig } from 'cypress';

const appName = process.env.NX_TASK_TARGET_PROJECT as string;
const outputFolderForProject = process.env.CI
  ? \`../../test-results/\${appName}\`
  : 'test-results';

export default defineConfig({
  e2e: { ...nxE2EPreset(__filename, { cypressDir: 'src' }) },
  reporter: '../../node_modules/cypress-multi-reporters',
  reporterOptions: {
    reporterEnabled: !process.env.CI
      ? 'spec'
      : 'spec, mocha-junit-reporter, mochawesome',
    mochaJunitReporterReporterOptions: {
      mochaFile: outputFolderForProject.concat(
        \`/downloads/junit-report/results-\${appName}-[hash].xml\`
      ),
    },
    mochawesomeReporterOptions: {
      charts: true,
      overwrite: false,
      html: false,
      json: true,
      reportDir: outputFolderForProject.concat('/downloads/reports-json-file'),
      reportFilename: '[name].html',
      embeddedScreenshots: true,
      inlineAssets: true,
    },
  },
});
"
`;

exports[`should run successfully with default options should update the support e2e file: apps/application-e2e/src/support/e2e.ts 1`] = `
"import './commands';
"
`;